react/features/settings/actions.web.ts

```diff
@@ -28,15 +33,40 @@ import {
     getShortcutsTabProps
 } from './functions.web';
 
+
 /**
  * Opens {@code LogoutDialog}.
  *
- * @param {Function} onLogout - The event in {@code LogoutDialog} that should be
- *  enabled on click.
+ * @param {string} logoutUrl - The url for logging out.
  * @returns {Function}
  */
-export function openLogoutDialog(onLogout: Function) {
-    return openDialog(LogoutDialog, { onLogout });
+export function openLogoutDialog(logoutUrl: string) {
+    return (dispatch: IStore['dispatch'], getState: IStore['getState']) => {
+        const state = getState();
+        const config = state['features/base/config'];
+        const { conference } = state['features/base/conference'];
+        const { jwt } = state['features/base/jwt'];
+
+        if (isTokenAuthEnabled(config) && config.tokenAuthUrlAutoRedirect && jwt) {
+
+            // user is logging out remove auto redirect indication
+            dispatch(setTokenAuthUrlSuccess(false));
+            dispatch(openDialog(LogoutDialog, {
+                onLogout() {
+                    if (browser.isElectron()) {
+                        const url = appendURLHashParam(logoutUrl, 'electron', 'true');
+
+                        window.open(url, '_blank');
+                        dispatch(hangup(true));
+                    } else {
+                        window.location.href = logoutUrl;
+                    }
+
+                    conference?.room.moderator.logout(() => dispatch(hangup(true)));
```

Comment:
This is for the non-token case.

Reply:
Sorry but I still think this is changed.

In old we can execute dispatch(setTokenAuthUrlSuccess(false)) and after that do redirect or moderator.logout.

While here if we do dispatch(setTokenAuthUrlSuccess(false)) there is nothing after that.